actions/login-to-gcs/action.yaml

```
@@ -0,0 +1,62 @@
+name: Login to GCS
+description: Composite action to login to Google Cloud Storage
+inputs:
+  bucket:
+    description: |
+      Name of bucket to upload to. Will default to grafanalabs-${repository.name}-${environment}
+    default: ""
+  environment:
+    description: |
+      Environment for uploading objects (can be either dev or prod).
+    default: dev
+
+outputs:
+  bucket:
+    description: "The name of the bucket that we have authenticated to."
+    value: ${{ steps.get-bucket-name.outputs.bucket }}
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" = "dev" ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == "prod" ]; then
+          PROJECT="grafanalabs-global"
+        else
+          echo "Invalid environment. Valid environment variable inputs: dev, prod"
+          exit 1
+        fi
```

suggestion: alternatively this could be written like so

```suggestion
      run: |
        case "${{ inputs.environment }}" in
          dev)
            PROJECT="grafanalabs-dev"
            ;;
          prod)
            PROJECT="grafanalabs-global"
            ;;
          *)
            echo "Invalid environment. Valid environment variable inputs: dev, prod"
            exit 1
            ;;
        esac
```

which might be a bit clearer to read - you decide :slightly_smiling_face:

---

I'm happy with how it is because it matches what's in the related `push-to-gar-docker` action. What I would actually like to see here is another shared workflow that both the `gcs` and `gar` can call so we don't have to duplicate code, but I'm afraid that might be a poor design choice given that there's a limit to how many actions can be invoked.

In any case, if we were going to go that route I would like to attack it in another PR.

---

righto!